web/modules/custom/collation_fixer/src/CollationFixer.php

```diff
@@ -0,0 +1,199 @@
+        "ALTER TABLE {$tableCollation->table} CHARACTER SET {$tableCollation->expectedCharset} COLLATE {$tableCollation->expectedCollation}"
+      );
+      if ($query instanceof StatementInterface) {
+        $result = $query->execute();
+        $status = $status && $result;
+      }
+
+      // Alter character set and collation of table data.
+      $query = $this->connection->query(
+        // Placeholders are intentionally not used there. It does not seem as
+        // if is it supported.
+        "ALTER TABLE {$tableCollation->table} CONVERT TO CHARACTER SET {$tableCollation->expectedCharset} COLLATE {$tableCollation->expectedCollation}"
+      );
+      if ($query instanceof StatementInterface) {
+        $result = $query->execute();
+        $status = $status && $result;
+      }
+    }
+
+    return $status;
+  }
+
+  /**
+   * Returns all available schemas ie. table names.
+   *
+   * @param ?string $table
+   *   A specific table name to check.
+   *
+   * @return string[]
+   *   An array of table names.
+   */
+  private function getSchema(?string $table = NULL): array {
+    $schemas = array_map(function (string $module) {
+      if ($this->moduleHandler->loadInclude($module, 'install')) {
+        return $this->moduleHandler->invoke($module, 'schema') ?? [];
+      }
+      else {
+        return [];
+      }
+    }, array_keys($this->moduleHandler->getModuleList()));
+    $moduleSchemas = array_merge(...$schemas);
+
+    $entitySchemas = [];
+    foreach ($this->entityTypeManager->getDefinitions() as $entityType) {
+      // Only list content entity types using SQL storage.
+      $entityStorage = $this->entityTypeManager->getStorage($entityType->id());
+      if ($entityStorage instanceof SqlEntityStorageInterface) {
+        foreach ($this->fieldManager->getFieldStorageDefinitions($entityType->id()) as $field) {
+          $entitySchemas = array_merge($entitySchemas, $entityStorage->getTableMapping()->getAllFieldTableNames($field->getName()));
+        }
+      }
+    }
+    $table_names = array_unique(array_merge(array_keys($moduleSchemas), $entitySchemas));
+
+    if (!empty($table)) {
+      return array_filter($table_names, function ($table_name) use ($table) {
+        return $table == $table_name;
+      });
+    }
+    else {
+      return $table_names;
+    }
+  }
+
+  /**
+   * Get the default charset to use if nothing else is specified.
+   */
+  private function getFallbackCharset(): string {
+    $connection_options = $this->connection->getConnectionOptions();
+    if (isset($connection_options['charset'])) {
+      $fallback_charset = $connection_options['charset'];
+    }
+    else {
+      // There does not seem to be a default charset used for Drupal 10.
+      // utf8mb4 was the charset suggested when multibyte UTF-8 support was
+      // added to Drupal 7. In lack of other sources we use this as well here.
+      // https://www.drupal.org/node/2754539
+      $fallback_charset = 'utf8mb4';
```

Comment:
Both fallback charset and fallback collation could be defined as a constant somewhere and maybe be configurable in the future?

Reply:
Witht the latest changes there is no such thing as a fallback collation.

If you expect the collation to have a certain value then module will compare this with the actual value to identify problems. Otherwise the module stays out of your way.